DataFormats/METReco/src/HcalNoiseHPD.cc

```diff
@@ -110,28 +110,28 @@ const edm::RefVector<HBHERecHitCollection> HcalNoiseHPD::recHits(void) const
   return rechits_;
 }
   
-float HcalNoiseHPD::recHitEnergy(float threshold) const
+float HcalNoiseHPD::recHitEnergy(const float threshold) const
 {
-  float total=0.0;
+  double total=0.0;
```

Comment:
Because it is useful to keep extended precision when accumulating an unknown number of terms in a summation. Helps with round-off errors.